pkg/common/cns-lib/vsphere/virtualcenter.go

```diff
@@ -306,7 +308,31 @@ func (vc *VirtualCenter) connect(ctx context.Context, requestNewSession bool) er
 		}
 	}
 	// If session has expired, create a new instance.
-	log.Warnf("Creating a new client session as the existing one isn't valid or not authenticated")
+	log.Infof("Creating a new client session as the existing one isn't valid or not authenticated")
+	if vc.Config.ReloadVCConfigForNewClient {
+		log.Info("reloading latest VC config from vSphere Config Secret")
+		cfg, err := config.GetConfig(ctx)
+		if err != nil {
+			return logger.LogNewErrorf(log, "failed to read config. Error: %+v", err)
+		}
+		var foundVCConfig bool
+		newVcenterConfigs, err := GetVirtualCenterConfigs(ctx, cfg)
+		if err != nil {
+			return logger.LogNewErrorf(log, "failed to get VirtualCenterConfigs. err=%v", err)
+		}
+		for _, newvcconfig := range newVcenterConfigs {
+			if newvcconfig.Host == vc.Config.Host {
+				newvcconfig.ReloadVCConfigForNewClient = true
+				vc.Config = newvcconfig
+				log.Infof("set latest VC config for vcenter: %q", vc.Config.Host)
```

Comment:
`set latest VC config for vcenter:` ==> `Successfully set latest VC config for vcenter:`

Reply:
done